drivers/hap_report/app/controllers/hap_report/warehouse_reports/hap_reports_controller.rb

```diff
@@ -0,0 +1,60 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HapReport::WarehouseReports
+  class HapReportsController < ApplicationController
+    include WarehouseReportAuthorization
+    before_action :set_report, only: [:show, :destroy, :details]
+
+    def index
+      @reports = report_scope.page(params[:page]).per(25)
+      @report = report_scope.build
+    end
+
+    def create
+      @report = report_scope.create(user_id: current_user.id, status: :pending, options: report_options)
+      @reports = report_scope.page(params[:page]).per(25)
+      if @report.valid?
+        @report.delay.build_report
```

Comment:
I think calling `.delay` bypasses some of our Job logic, it's probably fine, but it might be better to put this through `WarehouseReports::GenericReportJob`